AsyncDisplayKit/ASImageNode.mm

```diff
@@ -111,7 +111,9 @@ - (instancetype)initWithViewBlock:(ASDisplayNodeViewBlock)viewBlock
 - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 {
   ASDN::MutexLocker l(_imageLock);
-  if (_image)
+  if (CGSizeEqualToSize(self.preferredFrameSize, CGSizeZero) == NO)
+    return CGSizeMake(MIN(constrainedSize.width, self.preferredFrameSize.width), MIN(constrainedSize.height, self.preferredFrameSize.height));
```

Comment:
I think all ASDisplayNodes must respect the constrained size, so it might make sense to move this logic to ASDisplayNode. Then this method will look like this:

```
- (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
{
  ASDN::MutexLocker l(_imageLock);
  if (_image)
    return _image.size;
  else
    return [super calculateSizeThatFits:constrainedSize];
}
```